lib/WUI/link_content/prusa_link_api.cpp

```diff
@@ -33,6 +33,64 @@ namespace {
 
         return input.substr(prefix.size());
     }
+
+    optional<ConnectionState> parse_file_url(const RequestParser &parser, const size_t prefix_len, char *filename, const size_t filename_len) {
+        /*
+         * We do *not* use the uri with prefix removed. We need the safe
+         * transformation into a file name (removal of query params,
+         * forbidding of '..', etc).
+         */
+        /*
+         * TODO:
+         * We currently don't read the URI parameters. That is, we don't
+         * understand "?recursive=true" and therefore don't do recursive.
+         *
+         * The FileInfo handler doesn't know recursive yet either.
+         */
+        char fname[FILE_PATH_BUFFER_LEN + prefix_len];
+        const char *fname_real = fname + prefix_len;
+        if (parser.uri_filename(fname, sizeof fname)) {
+            size_t len = strlen(fname);
+            if (prefix_len > len) {
+                return StatusPage(Status::NotFound, parser.status_page_handling(), parser.accepts_json);
+            }
+
+            /*
+             * The octoprint API gives special meaning to /local and
+             * /sdcard. For us, everything lives in the USB (/usb). We
+             * remap these. Nevertheless, we never _generate_ these /local
+             * or such URIs, so we don't remap anything else but the "root".
+             */
+            static const char *const roots[] = {
+                "",
+                "/",
+                "/local",
+                "/local/",
+                "/sdcard",
+                "/sdcard/",
+            };
```

Comment:
Oops. I know it was my suggestion to re-use the code in the `/api/files`. But looking at it here, we probably don't want this particular re-mapping hack in the `/api/v1/files` ‒ one reason for creating the new API was to get rid of hardcoded `sdcard` and `local` and all the hacks related to that.

So I'd say we either want to somehow pass the list of mappings in as a parameter, or disable this part by some parameter.